lib/rack/request.rb

```diff
@@ -238,28 +239,26 @@ def xhr?
       end
 
       def host_with_port
-        if forwarded = get_header(HTTP_X_FORWARDED_HOST)
+        if forwarded = get_http_forwarded(:host)
+          forwarded.last
+        elsif forwarded = get_header(HTTP_X_FORWARDED_HOST)
```

Comment:
It's possible that `get_header(HTTP_X_FORWARDED_HOST) == ''` in some cases. For instance:
```sh
curl http://example.com -H 'X_FORWARDED_HOST;' -H 'HOST: example.com'
```
or
```
telnet example.com 80
Trying 127.0.0.1...
Connected to example.com.
Escape character is '^]'.
GET /HTTP/1.0
HOST: example.com
X_FORWARDED_HOST:


```